AFQ/_fixes.py

```diff
@@ -5,12 +5,28 @@
 from tqdm import tqdm
 from dipy.align import Bunch
 from dipy.tracking.local_tracking import LocalTracking
+from dipy.align.imaffine import AffineMap
 import random
 
 import sys
 import math
 
 
+class ConformedAffineMap(AffineMap):
+    """
+    Modifies AffineMap API to match functionality of DiffeomorphicMap.
+    Important for SLR maps to be indistinguishable from SYN maps
+    in terms of API.
+    """
+    def transform(self, *args, interpolation='linear', **kwargs):
+        kwargs['interp'] = interpolation
+        return super().transform_inverse(*args, **kwargs)
+
+    def transform_inverse(self, *args, interpolation='linear', **kwargs):
+        kwargs['interp'] = interpolation
+        return super().transform(*args, **kwargs)
```

Comment:
And `transform` here?

Reply:
For these two its really weird. As far as I can tell the DiffeomorphicMap and AffineMap have the opposite definitions of what is transform inverse and what is transform. Maybe I should just swap what is domain and codomain when I pass in, instead of swapping these transforms?

Reply:
Hmm. Could we raise this as an issue in DIPY? I saw that you already posted an issue about the API differences, but I think that it would be worth asking about this issue as well. It seems odd that this has the opposite directionality, and I don't know why that is. I want to make sure that we understand this correctly.

Reply:
For now, let's leave it as-is. It's being in the fixes module will (hopefully) help us remember that we need to eventually reconcile this.

Reply:
I will add a comment with an explanation, and also add the issue for dipy.

Reply:
OK. Looking further into dipy code, it seems like they should be doing the same transformation. Probably some problem on our end. I am going to double check everything again and run the code, see if I can isolate the problem.

Reply:
I figured out what was happening. When we load the DiffeomorphicMap we set it to inverse: https://github.com/yeatmanlab/pyAFQ/blob/822af52d43e960a84fa5ed0b831d97fdea75c7a7/AFQ/registration.py#L197 I am not really sure why. It makes all of our transforms transform_inverses and vice versa from then on out. That's why I had to do the same for the AffineMap to make it work. It seems to be necessary for the tests to make sense. @arokem, do you remember why we need to do this?

Reply:
No - I don't exactly remember why I set it this way. 

Let's see if  I understand, though: if we set this to `mapping.is_inverse = False` there, and then drop the code here, so that it does the same that DIPY currently does, would the tests then pass?

Reply:
Nope. mapping.is_inverse is for the DiffeomorphicMap, while above is for AffineMap. If we dropped both, we might have to change all of the transforms to transform_inverse in our code or something like that. I will look into this further.

Reply:
Maybe we can keep this fix for now, and then come back and clean this all up by moving to use [`nitransforms`](https://github.com/poldracklab/nitransforms) at some later point.

Reply:
See https://github.com/yeatmanlab/pyAFQ/issues/264

Reply:
Ok. I figured it out. It's very tricky and may deserve an issue report on Dipy.
The optimize function from SymmetricDiffeomorphicRegistration, which we use to get our Diffeomorphic map, 
returns with is_inverse set to true. This means, internally, transform are transform_inverse and vice versa. I think this flag is used in some important calculation in sdr.optimize, however, it should almost certainly be set back to false before the result is returned. This means, I think, that the transforms and transform_inverses in our code all need to be switched. We might want to talk all of this through on Monday.

Reply:
OK - for now, since we're not going to have an opportunity to discuss next week, let's leave this as-is for now, and then post an issue pointing to this conversation, so that we can come back to this later.